lab3/client/src/game.js

```diff
@@ -0,0 +1,33 @@
+// default interval as 1 second
+const interval = 1000;
+
+/**
+ * loop is main loop of the game, which will be executed once every second (
+ * based on the interval variable configuration)
+ */
+export function loop (store) {
+	// TODO: increment counter based on the generators in the state
+	// hint: read how many "generators" in store and iterate through them to
+	//       count how many value to increment to "resource"
+	// hint: remember to change event through `store.dispatch`
+	store.listeners.forEach(function(element){
+		store.dispatch({
+			type: constants.actions.INCREMENT,
+			payload: (element.rate * element.quantity)
+		})
+	})
+
+
+	// TODO: triggers stories from story to display state if they are passed
+	//       the `triggeredAt` points
+	// hint: use store.dispatch to send event for changing events state
+	store.listeners.forEach(function(element){
+		store.dispatch({
+			type: constants.actions.CHECK_STORY
+		})
+	})
```

Comment:
although this is functionally correct. You don't need to send CHECK_STORY per listener. The story has nothing to do with listener.